### PR TITLE
facebook authentication backend implementation

### LINKED ISSUE #1: Add Facebook authentication of login feature
Description: create Facdbook auth feature so that foodproviders can login with facebook account.

Accept criteria: testing account with Facebook account can login without filling signup form

### PR DESCRIPTION
**Pull request by foodie-dev-hongjun1**
aim to resolve #1 by adding facebook authentication into the backend

### CODE REVIEW

**Comment by foodie-dev-hongjun1** [comment]
failed heroku test because we did not setup payment for heroku CI test, since obviously we are as poor as Hongjun, hahahaha. So I disabled it - we will only test on circleci.

File: FoodieService/settings.py [hunk 1]
```diff
@@ -130,3 +135,35 @@
...
+AUTHENTICATION_BACKENDS = (
+   'rest_framework_social_oauth2.backends.DjangoOAuth2',
+   'django.contrib.auth.backends.ModelBackend',
+   # Facebook OAuth2
+   'social_core.backends.facebook.FacebookOAuth2',
+)
+
+# Facebook configuration
+SOCIAL_AUTH_FACEBOOK_KEY = '905660232906998'
+SOCIAL_AUTH_FACEBOOK_SECRET = 'beafa8cb81d5a24907e6455434056780'
```
Comment (foodie-dev-hongjun2) [inline FoodieService/settings.py]: The configs are hard coded for test purpose? Or we should just left them as is?

Comment (foodieBlackfoot) [inline FoodieService/settings.py]: yeah they are not supposed to be hard coded. But currently it is only for testing, so fine for now. And this facebook testing account has been locked 😂😂😂

Comment (foodie-dev-hongjun2) [inline FoodieService/settings.py]: OK, ship it!

**Pull request merged**